### PR TITLE
Refs #17906: Use 1.15+ boilerplate code

### PR DESCRIPTION
**Pull request by ekohl**
https://github.com/theforeman/smart-proxy/pull/492 adds the boilerplate to the smart proxy. This reduces the need for it to be copied into each plugin.

### CODE REVIEW

**Comment by ekohl** [comment]
This can now be merged. I suppose we should create a 1.13-stable branch as well.

File: test/dns_plugin_template_record_test.rb [hunk 1]
```diff
@@ -7,100 +7,76 @@ def setup
...
-    @provider.expects(:aaaa_record_conflicts).with('test.example.com', '2001:db8::1').returns(1)
-    assert_raise(Proxy::Dns::Collision) { @provider.create_aaaa_record('test.example.com', '2001:db8::1') }
+    #   MyService.expects(:create_reverse).with(:value => 'test.example.com', :name => '3.2.1.10.in-addr.arpa').returns(true)
+    assert @provider.do_create('3.2.1.10.in-addr.arpa', 'test.example.com', 'PTR')
   end
 
-  # Test PTR record creation
-  def test_create_ptr
+  # Test PTR record creation with an IPv6 address
+  def test_create_ptr_v4
```
Comment (domcleal) [inline test/dns_plugin_template_record_test.rb]: `ptr_v6`

**Review by domcleal — CHANGES_REQUESTED** [review]

File: test/dns_plugin_template_record_test.rb (hunk 1 above)
Comment (ekohl) [reply test/dns_plugin_template_record_test.rb]: Fixed

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @ekohl.

**Pull request merged**